core/include/detray/propagator/base_stepper.hpp

```diff
@@ -79,7 +106,7 @@ class base_stepper {
         }
 
         /// free track parameter
-        free_track_parameters_type _track;
+        typename free_track_parameters_type::vector_type _track;
```

Comment:
I am wondering why we want to expose everywhere how the track parameter is implemented internally by using the vector directly? Does the class not have a well defined interface for position, direction etc that we can hide behind how the data is actually saved and extracted with the ```track_helper```? 

It decreases maintainability to couple class internal details like this into surrounding code... For example, it would be nice if the ```track_helper``` does not appear outside the track parameter classes wherever possible, because if anything ever changes there, we have to go through and touch the code in all of those classes where it is used now